WORKSPACE

```diff
@@ -51,58 +79,69 @@ _piptool_install()
     # From https://pypi.python.org/pypi/grpcio/1.6.0
-    url = ("https://pypi.python.org/packages/c6/28/" +
-           "67651b4eabe616b27472c5518f9b2aa3f63beab8f62100b26f05ac428639/" +
-           "grpcio-1.6.0-cp27-cp27m-manylinux1_i686.whl"),
+    urls = [("https://pypi.python.org/packages/c6/28/" +
+             "67651b4eabe616b27472c5518f9b2aa3f63beab8f62100b26f05ac428639/" +
+             "grpcio-1.6.0-cp27-cp27m-manylinux1_i686.whl")],
 )
 
 http_file(
     name = "futures_3_1_1_whl",
+    downloaded_file_path = "futures-3.1.1-py2-none-any.whl",
     sha256 = "c4884a65654a7c45435063e14ae85280eb1f111d94e542396717ba9828c4337f",
     # From https://pypi.python.org/pypi/futures
-    url = ("https://pypi.python.org/packages/a6/1c/" +
-           "72a18c8c7502ee1b38a604a5c5243aa8c2a64f4bba4e6631b1b8972235dd/" +
-           "futures-3.1.1-py2-none-any.whl"),
+    urls = [("https://pypi.python.org/packages/a6/1c/" +
+             "72a18c8c7502ee1b38a604a5c5243aa8c2a64f4bba4e6631b1b8972235dd/" +
+             "futures-3.1.1-py2-none-any.whl")],
 )
 
 http_file(
     name = "futures_2_2_0_whl",
+    downloaded_file_path = "futures-2.2.0-py2.py3-none-any.whl",
     sha256 = "9fd22b354a4c4755ad8c7d161d93f5026aca4cfe999bd2e53168f14765c02cd6",
     # From https://pypi.python.org/pypi/futures/2.2.0
-    url = ("https://pypi.python.org/packages/d7/1d/" +
-           "68874943aa37cf1c483fc61def813188473596043158faa6511c04a038b4/" +
-           "futures-2.2.0-py2.py3-none-any.whl"),
+    urls = [("https://pypi.python.org/packages/d7/1d/" +
+             "68874943aa37cf1c483fc61def813188473596043158faa6511c04a038b4/" +
+             "futures-2.2.0-py2.py3-none-any.whl")],
 )
 
 http_file(
     name = "mock_whl",
+    downloaded_file_path = "mock-2.0.0-py2.py3-none-any.whl",
     sha256 = "5ce3c71c5545b472da17b72268978914d0252980348636840bd34a00b5cc96c1",
     # From https://pypi.python.org/pypi/mock
-    url = ("https://pypi.python.org/packages/e6/35/" +
-           "f187bdf23be87092bd0f1200d43d23076cee4d0dec109f195173fd3ebc79/" +
-           "mock-2.0.0-py2.py3-none-any.whl"),
+    urls = [("https://pypi.python.org/packages/e6/35/" +
+             "f187bdf23be87092bd0f1200d43d23076cee4d0dec109f195173fd3ebc79/" +
+             "mock-2.0.0-py2.py3-none-any.whl")],
 )
 
 http_file(
     name = "google_cloud_language_whl",
+    downloaded_file_path = "google_cloud_language-0.29.0-py2.py3-none-any.whl",
     sha256 = "a2dd34f0a0ebf5705dcbe34bd41199b1d0a55c4597d38ed045bd183361a561e9",
     # From https://pypi.python.org/pypi/google-cloud-language
-    url = ("https://pypi.python.org/packages/6e/86/" +
-           "cae57e4802e72d9e626ee5828ed5a646cf4016b473a4a022f1038dba3460/" +
-           "google_cloud_language-0.29.0-py2.py3-none-any.whl"),
+    urls = [("https://pypi.python.org/packages/6e/86/" +
+             "cae57e4802e72d9e626ee5828ed5a646cf4016b473a4a022f1038dba3460/" +
+             "google_cloud_language-0.29.0-py2.py3-none-any.whl")],
```

Comment:
You need a hygenic macro to create these.

Reply:
You mean to factor out the whl name from the url and the `downloaded_file_path`? I'm not sure what value the macro adds besides code golfing -- What would the macro's arguments be, and what would the URL-minus-the-wheel-name be? Or would you suggest parsing the URL to determine `downloaded_file_path`?

I'm not even sure I like the downloaded file path being the wheel name as opposed to the default string, `<repo name>/file/downloaded`, but I kept it this way for compatibility with the existing tests. If I create a macro, that risks users loading it and adopting that convention without us having really thought it out.

Reply:
Okay, that makes sense. Leave it as-is.